src/Core/PackageKitBackend.vala

```diff
@@ -474,8 +474,17 @@ public class AppCenterCore.PackageKitBackend : Backend, Object {
 
         Pk.Exit exit_status = Pk.Exit.UNKNOWN;
         string[] packages_ids = {};
-        foreach (var pk_package in package.change_information.updatable_ids) {
-            packages_ids += pk_package;
+        foreach (var pk_package in package.change_information.updatable_packages) {
```

Comment:
I wonder if it would make sense to either sort `updateable_packages` or split the packages for different backends to avoid looping through all of them in each backends but I guess that's not the focus of this PR.

Reply:
@donadigo That was a really good point, thanks! I've swapped out the data structure for a MultiMap, so now we only iterate over the packages belonging to the backend in question.